### PR TITLE
docs: add a section for Node.js module

### PR DESCRIPTION
**Pull request by eunjae-lee**
This PR adds a section for Node.js module.

[View Rendered](https://github.com/algolia/search-insights.js/blob/bfa3ffd0f36f494f349db19b86a5d2b7772b1d26/README.md)

### CODE REVIEW

**Comment by Haroenv** [comment]
Doc itself looks ✅, but I wouldn't merge this to the readme which is visible to the user until we published. It could be to a temporary "node" branch?

**Comment by eunjae-lee** [comment]
> Doc itself looks ✅, but I wouldn't merge this to the readme which is visible to the user until we published. It could be to a temporary "node" branch?

Right. That's why I've labeled 'DO NOT MERGE'.

File: README.md [hunk 1]
```diff
@@ -204,6 +235,36 @@ To run all examples and play around with the code you have to run two separate c
 - `yarn dev` - runs webpack and node dev server
 - `yarn build:dev` - runs rollup in watch mode - livereload if you do changes to the insights library
 
+## Node.js module
+
+Insights library can be used on backend as a Node.js module.
```
Comment (tkrugg) [inline README.md]:
```suggestion
Insights library can be used on the backend as a Node.js module.
```

File: README.md [hunk 2]
```diff
@@ -204,6 +235,36 @@ To run all examples and play around with the code you have to run two separate c
...
+  appId: "APPLICATION_ID",
+  apiKey: "SEARCH_API_KEY"
+});
+```
+
+It exports the same `aa` interface.
+
+### setUserToken
+
+On node environment, unlike browser environment, `setUserToken` must be called before sending any event.
```
Comment (tkrugg) [inline README.md]:
```suggestion
On the node environment, unlike the browser environment, `setUserToken` must be called before sending any event.
```
maybe it's just my frenglish that makes me add `the` everywhere ? 🤓 
maybe @Haroenv can confirm?

Comment (eunjae-lee) [reply README.md]: maybe it's just my konglish not using `the` enough.

File: README.md [hunk 3]
```diff
@@ -204,6 +235,36 @@ To run all examples and play around with the code you have to run two separate c
 - `yarn dev` - runs webpack and node dev server
 - `yarn build:dev` - runs rollup in watch mode - livereload if you do changes to the insights library
 
+## Node.js module
```
Comment (tkrugg) [inline README.md]: I'm not sure I would have put this in an entirely new section.
The main difference is in the installation process right? 
- in the browser, we use the snippet
- in node, we use whatever dependency management we have

So maybe this should just be a section under `Loading and initializing the library` ?

Comment (eunjae-lee) [reply README.md]: Right. I will move this part under `Loading and ...` and also add how to `npm install` or `yarn install` this.

File: README.md (hunk 2 above)
Comment (Haroenv) [reply README.md]: your suggestions seem better @tkrugg

File: README.md [hunk 4]
```diff
@@ -39,6 +68,38 @@ search analytics.
...
+  appId: "APPLICATION_ID",
+  apiKey: "SEARCH_API_KEY"
+});
+```
+
+It exports the same `aa` interface.
+
+##### setUserToken
+
+On the Node.js environment, unlike the browser environment, `setUserToken` must be called before sending any event.
```
Comment (tkrugg) [inline README.md]: For node I think most user will want to just include the userToken in the event itself. 
IMHO
- If you're seeding events, this is more practical
- if you're sending events from a route, this more practical too

```js
aa("clickedObjectIDs", {
   userToken: "xxx",
   /* ... */
})
```
We should at least document this other way of including the user token.

**Comment by eunjae-lee** [comment]
merged `develop` into here

File: README.md [hunk 5]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
+```bash
+$ npm install search-insights
+or
+$ yarn add search-insights
+```
+
+#### 2. Initialize the library
+
+```js
+const aa = require("search-insights");
```
Comment (francoischalifour) [inline README.md]:
```suggestion
const aa = require("search-insights");

```

File: README.md [hunk 6]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
 | `region`          | `'de' \| 'us'` | Automatic                | The DNS server to target                       |
 | `cookieDuration`  | `number`       | `15552000000` (6 months) | The cookie duration in milliseconds            |
 
+### On Node.js
+
+#### 1. Install the library
+
+Insights library can be used on the backend as a Node.js module. _(Node.js `>= 8.16.0` required)_
```
Comment (francoischalifour) [inline README.md]: The Node.js requirement mention can go in the previous section "On Node.js".

File: README.md [hunk 7]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
+
+```bash
+$ npm install search-insights
+or
```
Comment (francoischalifour) [inline README.md]:
```suggestion
# or
```

File: README.md [hunk 8]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
 | `cookieDuration`  | `number`       | `15552000000` (6 months) | The cookie duration in milliseconds            |
 
+### On Node.js
+
+#### 1. Install the library
+
+Insights library can be used on the backend as a Node.js module. _(Node.js `>= 8.16.0` required)_
+
+```bash
+$ npm install search-insights
```
Comment (francoischalifour) [inline README.md]: Removing `$` makes it easier to copy paste.

File: README.md [hunk 9]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
+or
+$ yarn add search-insights
+```
+
+#### 2. Initialize the library
+
+```js
+const aa = require("search-insights");
+aa("init", {
```
Comment (francoischalifour) [inline README.md]: To be consistent, we can use single quotes everywhere.

File: README.md [hunk 10]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
+  appId: "APPLICATION_ID",
+  apiKey: "SEARCH_API_KEY"
+});
+```
+It exports the same `aa` interface.
```
Comment (francoischalifour) [inline README.md]: Not sure this is needed.

File: README.md [hunk 11]
```diff
@@ -40,6 +58,40 @@ aa('setUserToken', 'USER_ID');
...
+
+#### Add `userToken`
+
+On the Node.js environment, unlike the browser environment, `userToken` must be specified when sending any event.
+
+```js
+aa("clickedObjectIDs", {
+  userToken: "id-of-user",
```
Comment (francoischalifour) [inline README.md]: All other mentions follow this kind of placeholder.

```suggestion
  userToken: "USER_ID",
```

File: README.md [hunk 12]
```diff
@@ -5,9 +5,27 @@
...
+- [Examples](#examples)
+- [Contributing](#contributing)
+- [License](#license)
+
+<!-- tocstop -->
+
 ## Getting started
 
-### 1. <a name="loading"></a>Load the library
+### On Browser
```
Comment (francoischalifour) [inline README.md]: We can remove "On" IMO.

**Comment by eunjae-lee** [comment]
Thanks @francoischalifour. I've applied all your feedbacks (which I agree with).

File: README.md [hunk 13]
```diff
@@ -5,9 +5,27 @@
 
 Search Insights lets you report click, conversion and view metrics using the [Algolia Insights API](https://www.algolia.com/doc/rest-api/insights/#overview).
 
+## Table of Contents
+
+<!-- toc -->
+
+- [Getting started](#getting-started)
+  - [On Browser](#on-browser)
```
Comment (francoischalifour) [inline README.md]: The TOC needs to be updated and we're good!

Comment (eunjae-lee) [reply README.md]: Good point!

**Pull request merged**